framework/db/Query.php

```diff
@@ -590,7 +590,7 @@ public function addSelect($columns)
         if ($this->select === null) {
             $this->select = $columns;
         } else {
-            $this->select = array_merge($this->select, $columns);
+            $this->select = array_unique(array_merge($this->select, $columns), SORT_REGULAR);
```

Comment:
`array_unique()` changes order - this will probably break BC.

Reply:
Yes. Order should be kept and it should be in the test.

Reply:
As I can see, array_unique does not changes order of elements. It just deletes duplicates. I'm not good at C++, but I think [this code block](https://github.com/php/php-src/blob/a8a17a72b06e6ea39a70eaf45ad2c3d6668e29ba/ext/standard/array.c#L4527) creates internal array of pointers and sorts it for comparation. After that it deletes elements from the original array so it's not being sorted at all. This snippets tests functionality: https://3v4l.org/2v8UV It saved all unique elements from first array and added unique elements from the second one.

Reply:
@wapmorgan You're right. But you should probably leave `SORT_STRING` as default `$sort_flags` - `SORT_REGULAR` is unreliable with when you compare different types: https://3v4l.org/coIXF